packages/app/src/components/approve/ApproveToken.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, MouseEventHandler } from "react";
+import { Button, Container, Tooltip, OverlayTrigger, Spinner } from "react-bootstrap";
+import { constants, Contract } from "ethers";
+import { addresses } from "@tender/contracts";
+import { useContractFunction } from "@usedapp/core";
+
+type Props = {
+  protocolName: string;
+  tokenSymbol: string;
+  isTokenApproved: boolean;
+  tokenAddress: Contract;
+};
+
+const ApproveToken: FC<Props> = ({ tokenSymbol, protocolName, isTokenApproved, tokenAddress }) => {
```

Comment:
I don't think we need protocolName here, but symbol can be useful for UX (and doesn't have to be the protocol symbol but is whatever symbol of the token being approved e.g; tLPT, LPT or tLPT/LPT-Liquidity-pool-token). 

I think the props should be 

```
symbol: string; 
hasAllowance: boolean;
spender: string;
amount: BigNumberish (in this case not needed since we'll use `constants.MaxUint256`) 
```